### PR TITLE
Add c-n, c-j, c-p, c-k, c-m to console keybind

### PR DESCRIPTION
**Pull request by heavenshell**
Hi, first thank you for create nice and useful extension!

I add following behaviors to Vim-vixen.

- `c-n`, ~~`c-j`~~ for select next item.
- `c-p`, ~~`c-k`~~ for select previous item.
- `c-m` for select item.

These console keybinds are same as Vim(Vimperator)'s completion selector.

Please check and review this and it's very pleasure to give me advices.

Thank you.

### CODE REVIEW

**Comment by idlewan** [comment]
This is great, I was wondering how to use that list, I didn't notice that you had to use the tab key.
In addition to all of these, the up/down arrow keys should probably also be working to select lines.

**Comment by heavenshell** [comment]
>In addition to all of these, the up/down arrow keys should probably also be working to select lines.

ah, right. I'll add later 👍

**Comment by heavenshell** [comment]
Add `up/down arrow keys` for select next/prev item ☺️

File: src/console/components/console.js [hunk 1]
```diff
@@ -55,25 +71,25 @@ export default class ConsoleComponent {
       break;
     case KeyboardEvent.DOM_VK_M:
       if (e.ctrlKey) {
-        e.stopPropagation();
-        e.preventDefault();
-        return this.onEntered(e.target.value);
+        this.doEnter(e);
```
Comment (idlewan) [inline src/console/components/console.js]: Does this need a `return` as well?

Comment (heavenshell) [reply src/console/components/console.js]: nice catch! thx!

Comment (heavenshell) [reply src/console/components/console.js]: oh, I missed your fixed comment :-p
Original code have `return`.
I think we should confirm to @ueokande .

**Comment by ueokande** [comment]
@heavenshell Thanks for your PR.
Your patch is not acceptable.  I agree to use `<C-N>` and `<C-P>` to focus next and previous items and `<C-M>` to select item.  They are also used in Vim.
In my experience, `<C-J>`, `<C-K>`, and arrow keys have not been commonly used for select items in console. What is a reason for tha decision to use these keys?

**Comment by heavenshell** [comment]
@ueokande Thank you for review my PR!

> In my experience, <C-J>, <C-K>, and arrow keys have not been commonly used for select items in console. What is a reason for tha decision to use these keys?

`arrow key` is also Vim's completion selector behavior. So I added.
```
$ vim --clean
i<c-x><c-f><arrow down>
```

`<c-j>, <c-k>` are not Vim's default behavior but at Vimperator's behavior.
But I can understand `<c-j>, <c-k>` is not common behavior.
So, if I drop `<c-j>, <c-k>`, could you accept this?

**Comment by heavenshell** [comment]
Drop `<C-j>`, `<C-k>` 😊

**Comment by ueokande** [comment]
Arrow keys seem completion from *command-line history* in Vim. I have a plan to implement is in the future.  Please drop arrow keys from key maps.

**Comment by heavenshell** [comment]
@ueokande 

>Arrow keys seem completion from command-line history in Vim. I have a plan to implement is in the future. Please drop arrow keys from key maps.

This is so nice and make sense!
Drop arrow keys from console 👍

**Comment by ueokande** [comment]
@heavenshell In my environment (Linux, Firefox 58 developer edition), `<C-N>` does not work to select next item in completion.  The new tab is opened by Firefox's key binding.  How do I solve it?

**Comment by heavenshell** [comment]
@ueokande Thank you for telling me this.
Oh, I usally use macOS and did't notice about confilict with Firefox's keybind.

It's better to do followings
 - if platform is Linux and state is opening console than ignore OS's default keybinds

IMHO, this is very difficult(I'm sorry, I'm WebExtensions newbie).

I'll continue to investigate this, but do you have any idea?
Sorry for bothering you...

P.S.
I found `Shift-Ctrl-n` move next at Linux(Ubuntu) + Firefox57...

**Comment by ueokande** [comment]
@heavenshell 
I understood a situation on `<C-N>` binding issue.  Some Firefox's short-cut keys are not overrided by WebExtensions.  It is already reported in Bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1320332

I would merge your PR, and register known problem in issues.  I wish Mozilla will resolve it. 🙏 
Thank you.

**Comment by ueokande** [comment]
#200

**Comment by heavenshell** [comment]
Thank you so much including this!!

**Pull request merged**